tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -159,100 +166,117 @@ public void close() throws Exception {
     @ParameterizedTest(name = ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME)
     @ValueSource(strings = {"zk", "kraft"})
     public void testCreate(String quorum) throws Exception {
-        createAndWaitTopic(buildTopicCommandOptionsWithBootstrap(
-            "--create", "--partitions", "2", "--replication-factor", "1", "--topic", testTopicName));
-
-        assertTrue(adminClient.listTopics().names().get().contains(testTopicName));
+        TestUtils.createTopicWithAdmin(adminClient, testTopicName, scalaBrokers, scalaControllers, 2, 1,
+                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
```

Comment:
Could you explain what's the difference between the existing `createAndWaitTopic` and `createTopicWithAdmin` util methods?

Reply:
Also, if we don't use `createAndWaitTopic` and any other util methods in this class, should we also remove them?

Reply:
`createAndWaitTopic` creates the topic and wait for a valid leader to be propagated to metadatacache (it calls the `TestUtils` API `waitForPartitionMetadata`. Meanwhile, `createTopicWithAdmin` calls the `TestUtils` API `waitForAllPartitionsMetadata` and wait until the expected number of partitions is in the metadata cache in each broker (And it also calls `ensureConsistentKRaftMetadata` which wait until the kraft broker metadata have caught up to the controller) 

I believe the use of `createTopicWithAdmin` will lead to more stable test result...What do you think ? Happy to hear your thoughts

Reply:
Thanks for the explanation. I think both have similar results, but if there are already `TestUtils` methods, we should use them instead of creating a duplicated one. Please remove unused methods in the test class. Thanks.

Reply:
Thanks. I've removed the unused methods in the test class